scripts/test.sh

```diff
@@ -124,126 +133,63 @@ start_packager() {
-        cat "$log_temp"
-        printf "\n\n*** End Xcode Failure ***\n"
-        exit $EXITCODE
-    }
-    rm "$log_temp"
+  
+  echo "Installing application on ${SIM_DEVICE_NAME}"
+  echo "Application Path" $(pwd)/build/Build/Products/$CONFIGURATION-iphonesimulator/$1.app
+  xcrun simctl install ${SIM_DEVICE_NAME} $(pwd)/build/Build/Products/$CONFIGURATION-iphonesimulator/$1.app
+  
+  
+  echo "Launching application. (output is in $(pwd)/build/out.txt)"
+  xcrun simctl launch --console ${SIM_DEVICE_NAME} io.realm.$1 | tee $(pwd)/build/out.txt
+
+  echo "Shuttting down ${SIM_DEVICE_NAME} simulator. (device is not deleted. you can use it to debug the app)"
+  shutdown_ios_simulator
+
+  echo "Checking tests results"
+  if grep -q "REALM_FAILING_TESTS" $(pwd)/build/out.txt; then
+      echo "*** REALM JS TESTS FAILED. See tests results above ***"
+      exit 20
   else
-    xcrun xcodebuild -scheme "$1" -configuration "$CONFIGURATION" -sdk iphonesimulator -destination id="$IOS_SIM_DEVICE" test || {
-        EXITCODE=$?
-        echo "*** Failure (exit code $EXITCODE). ***"
-        exit $EXITCODE
-    }
+      echo "*** $1 SUCCESS ***"
   fi
 }
 
-setup_ios_simulator() {
-  # - Ensure one version of xcode is chosen by all tools
-  if [[ -z "$DEVELOPER_DIR" ]]; then
-    DEV_DIR="$(xcode-select -p)"
-    export DEVELOPER_DIR=$DEV_DIR
-  fi
-
-  # -- Ensure that the simulator is ready
-
-  if [ $CI_RUN == true ]; then
-    # - Kill the Simulator to ensure we are running the correct one, only when running in CI
-    echo "Resetting simulator using toolchain from: $DEVELOPER_DIR"
+setup_ios_simulator() { 
+  #try deleting old simulator with same name.
+  echo "Preparing to create a new simulator"
+  delete_ios_simulator >/dev/null 2>&1
 
-    # Quit Simulator.app to give it a chance to go down gracefully
-    local deadline=$((SECONDS+5))
-    while pgrep -qx Simulator && [ $SECONDS -lt $deadline ]; do
-      osascript -e 'tell app "Simulator" to quit without saving' || true
-      sleep 0.25 # otherwise the pkill following will get it too early
-    done
-
-    # stop CoreSimulatorService
-    launchctl remove com.apple.CoreSimulator.CoreSimulatorService 2>/dev/null || true
-    sleep 0.25 # launchtl can take a small moment to kill services
-
-    # kill them with fire
-    while pgrep -qx Simulator com.apple.CoreSimulator.CoreSimulatorService; do
-      pkill -9 -x Simulator com.apple.CoreSimulator.CoreSimulatorService || true
-      sleep 0.05
-    done
-
-    # - Prod `simctl` a few times as sometimes it fails the first couple of times after switching XCode vesions
-    local deadline=$((SECONDS+5))
-    while [ -z "$(xcrun simctl list devices 2>/dev/null)" ] && [ $SECONDS -lt $deadline ]; do
-      : # nothing to see here, will stop cycling on the first successful run
-    done
+  #parse devices
+  IOS_RUNTIME=$(xcrun simctl list runtimes |  grep -m1 -o '(com.apple.CoreSimulator.SimRuntime.iOS.*)' | sed 's/[()]//g')
```

Comment:
It doesn't work for me locally. Doing `xcrun simctl list runtimes |  grep -m1 'com.apple.CoreSimulator.SimRuntime.iOS.*' | awk '{ print $2 }'`, gets the iOS version number for me locally. When running `./scripts/test.sh react-tests` I see the following in the end:

```
Shuttting down realm-js-test simulator. (device is not deleted. you can use it to debug the app)
Checking tests results
*** ReactTests SUCCESS ***
```

Reply:
The ending is normal